categorize_notify/program.py

```diff
@@ -0,0 +1,162 @@
+from autokitteh import google, slack
+from openai import OpenAI
+
+
+GMAIL_CONNECTION_NAME = "my_gmail"
+OPENAI_CONNECTION_NAME = "my_chatgpt"
+SLACK_CHANNELS = ["demos", "engineering", "ui"]
+SLACK_CONNECTION_NAME = "my_slack"
+
+
+def on_http_get(data):
+    total_messages = None
+    while True:
+        total_messages = _poll_inbox(total_messages)
+        time.sleep(10)
+
+
+@autokitteh.activity
+def _poll_inbox(prev_total_messages: int):
+    gmail = google.gmail_client(GMAIL_CONNECTION_NAME).users()
+    # TODO: fix: If user deletes emails before the next poll then new email messages will be missed.
+    curr_total_messages = _get_message_count(gmail)
+    if prev_total_messages and curr_total_messages > prev_total_messages:
+        new_email_count = curr_total_messages - prev_total_messages
+        message_ids = _get_latest_message_ids(gmail, new_email_count)
+        for message_id in message_ids:
+            _process_email(gmail, message_id)
+
+    return curr_total_messages
+
+
+def _process_email(gmail, message_id: str):
+    message = gmail.messages().get(userId="me", id=message_id).execute()
+    email_content = _parse_email(message)
+    if email_content:
+        channel = _categorize_email(email_content, SLACK_CHANNELS)
+        if channel:
+            client = slack.slack_client(SLACK_CONNECTION_NAME)
+            client.chat_postMessage(channel=channel, text=email_content)
+
+        # Add label to email
+        label_id = _get_label_id(gmail, channel)
+        if not label_id:
+            created_label = _create_label(gmail, channel)
+            label_id = created_label["id"]
+        body = {"addLabelIds": [label_id]}
+        gmail.messages().modify(userId="me", id=message_id, body=body).execute()
+
+
+def _get_latest_message_ids(gmail, new_email_count: int):
+    """Get the latest email message_id from the user's inbox.
+    Args:
+        gmail: An authorized Gmail API service instance.
+        new_email_count: The number of new email messages to retrieve.
+    Returns: A list of email message_id strings.
+    """
+    results = gmail.messages().list(userId="me", maxResults=new_email_count).execute()
+
+    return [msg["id"] for msg in results.get("messages", [])]
+
+
+def _parse_email(message: dict):
+    """Parse provided email.
+    Args:
+        message: The email message to parse
+    Returns: The email body as a string.
+    """
+    payload = message["payload"]
+    for part in payload.get("parts", []):
+        if part["mimeType"] == "text/plain":
+            return base64.urlsafe_b64decode(part["body"]["data"]).decode("utf-8")
+
+
+def _create_label(gmail, label_name: str) -> dict:
+    """Create a new label in the user's gmail account.
+    Args:
+        gmail: An authorized Gmail API service instance.
+        label_name: The name of the label to be created.
+    Returns: The created label as a dictionary.
+    https://developers.google.com/gmail/api/reference/rest/v1/users.labels#Label
```

Comment:
If you want to add notes/links/etc to the docstring, do it in (a) separate paragraph(s) between the first line and "Args".

Reply:
Done. I removed the docstring. I think it's pretty obvious what's happening.